universal-application-tool-0.0.1/app/repository/VersionRepository.java

```diff
@@ -101,6 +108,31 @@ public Version getDraftVersion() {
     }
   }
 
+  private void handleMultipleDrafts() {
```

Comment:
Thoughts (non-blocking): I wonder if these should be guarded by a single transaction.

Reply:
It's hard because the transactions are global - they probably SHOULD but you can't be sure you haven't arrived here inside another transaction, and `commitTransaction` might prematurely commit the parent transaction.

I _think_.  no idea if my experiments really prove this at all.